internal/statestore/backfill_test.go

```diff
@@ -0,0 +1,344 @@
+		description     string
+		backfillID      string
+		expectedCode    codes.Code
+		expectedMessage string
+	}{
+		{
+			description:     "backfill is found",
+			backfillID:      "mockBackfillID",
+			expectedCode:    codes.OK,
+			expectedMessage: "",
+		},
+		{
+			description:     "empty id passed, err expected",
+			backfillID:      "",
+			expectedCode:    codes.NotFound,
+			expectedMessage: "Backfill id:  not found",
+		},
+		{
+			description:     "wrong id passed, err expected",
+			backfillID:      "123456",
+			expectedCode:    codes.NotFound,
+			expectedMessage: "Backfill id: 123456 not found",
+		},
+		{
+			description:     "item of a wrong type is requested, err expected",
+			backfillID:      "wrong-type-key",
+			expectedCode:    codes.Internal,
+			expectedMessage: "failed to unmarshal the backfill proto, id: wrong-type-key: proto: can't skip unknown wire type",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.description, func(t *testing.T) {
+			backfillActual, errActual := service.GetBackfill(ctx, tc.backfillID)
+			if tc.expectedCode == codes.OK {
+				require.NoError(t, errActual)
+				require.NotNil(t, backfillActual)
+				require.Equal(t, expectedBackfill.Id, backfillActual.Id)
+				require.Equal(t, expectedBackfill.SearchFields, backfillActual.SearchFields)
+				require.Equal(t, expectedBackfill.Extensions, backfillActual.Extensions)
+				require.Equal(t, expectedBackfill.Generation, backfillActual.Generation)
+			} else {
+				require.Error(t, errActual)
+				require.Equal(t, tc.expectedCode.String(), status.Convert(errActual).Code().String())
+				require.Contains(t, status.Convert(errActual).Message(), tc.expectedMessage)
+			}
+		})
+	}
+
+	// pass an expired context, err expected
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	service = New(cfg)
+	res, err := service.GetBackfill(ctx, "12345")
+	require.Error(t, err)
+	require.Nil(t, res)
+	require.Equal(t, codes.Unavailable.String(), status.Convert(err).Code().String())
+	require.Contains(t, status.Convert(err).Message(), "GetBackfill, id: 12345, failed to connect to redis:")
+}
+
+func TestDeleteBackfill(t *testing.T) {
+	cfg, closer := createRedis(t, false, "")
+	defer closer()
+	service := New(cfg)
+	require.NotNil(t, service)
+	defer service.Close()
+	ctx := utilTesting.NewContext(t)
+
+	err := service.CreateBackfill(ctx, &pb.Backfill{
+		Id:         "mockBackfillID",
+		Generation: 1,
+	})
+	require.NoError(t, err)
+
+	var testCases = []struct {
+		description     string
+		ticketID        string
+		expectedCode    codes.Code
+		expectedMessage string
+	}{
+		{
+			description:     "backfill is found and deleted",
+			ticketID:        "mockBackfillID",
+			expectedCode:    codes.OK,
+			expectedMessage: "",
+		},
+		{
+			description:     "empty id passed, no err expected",
+			ticketID:        "",
+			expectedCode:    codes.OK,
+			expectedMessage: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
```

Comment:
what is this convention for??

Reply:
This is an agreement to avoid late binding of tc variable in case of a closure in a for loop.
It's not a case for this particular loop, but we agreed with @Laremere to use such a pattern everywhere.

Reply:
I see one of the answers could be found here:
https://github.com/kyoh86/scopelint/issues/4#issuecomment-471661062